spec/system/log_in_system_spec.rb

```diff
@@ -17,8 +17,35 @@
       fill_in "user_email", with: "deactivated@example.com"
       fill_in "user_password", with: "password"
       find('input[name="commit"]').click
-
       expect(page).to have_content("Invalid Email or password")
     end
   end
-end
+
+  example "Check for modal in login page for staging env" do
+    allow(Rails.env).to receive_message_chain(:staging?).and_return(true)
+    # allow(Rails).to receive_message_chain(:env, :staging?).and_return(true)
+    visit "/users/sign_in"
+    find('#warningModal')
+    expect(page).to have_content 'This site is for TEST purposes only!'
+  end
+
+  example "Check for modal in password reset page for staging env" do
+    allow(Rails.env).to receive_message_chain(:staging?).and_return(true)
+    # allow(Rails).to receive_message_chain(:env, :staging?).and_return(true)
+    visit "/users/password/new"
+    find('#warningModal')
+    expect(page).to have_content 'This site is for TEST purposes only!'
+  end
+
+  example "Check there is no modal in login page for non staging env" do
+    allow(Rails.env).to receive_message_chain(:staging?).and_return(false)
```

Comment:
In this case you can use `receive` instead of `receive_message_chain` if you are only stubbing a single method. The time you'd use `receive_message_chain` is when you want to stub the value of a method chain.

Reply:
got it! all this stubbing stuff is starting to click (: thank you!